runtime/bcverify/rtverify.c

```diff
@@ -1165,15 +1165,50 @@ verifyBytecodes (J9BytecodeVerificationData * verifyData)
 				break;
 
 			case JBreturn1:
+			case JBreturnB:
+			case JBreturnC:
+			case JBreturnS:
+			case JBreturnZ:
 			case JBsyncReturn1:
 				if (returnChar) {
+					U_32 returnType = 0;
 					/* single character return description */
 					if ((returnChar < 'A') || (returnChar > 'Z')) {
 						inconsistentStack = TRUE;
 						goto _illegalPrimitiveReturn;
 					}
 					type = (UDATA) oneArgTypeCharConversion[returnChar - 'A'];
 					POP_TOS(type);
+
+					/* check methods that return char, byte, short, or bool use the right opcode */
+					if (BCV_BASE_TYPE_INT == type){
+						switch(bc){
+						case JBreturnB:
+							returnType = BCV_BASE_TYPE_BYTE_BIT;
+							break;
+
+						case JBreturnC:
+							returnType = BCV_BASE_TYPE_CHAR_BIT;
+							break;
+
+						case JBreturnS:
+							returnType = BCV_BASE_TYPE_SHORT_BIT;
+							break;
+
+						case JBreturnZ:
+							returnType = BCV_BASE_TYPE_BOOL_BIT;
+							break;
+
+						case JBreturn1:
+						case JBsyncReturn1:
+							returnType = BCV_BASE_TYPE_INT_BIT;
+							break;
+						default:
+							//TODO mark unreachable
```

Comment:
Can you add an Assert unreachable here?

Reply:
Do we have a common solution for that? I tried doing a quick search of the code and It looks like its usually a module or function specific tracepoint. Wasn't sure if there was a more generic solution, hence the `TODO`

Reply:
> usually a module or function specific tracepoint

That's the generic solution :)  Most (many?) tdf files have the basic asserts in them at this point.  Copying over an existing one if it isn't present already should be fairly straightforward